core/src/solve.jl

```diff
@@ -449,6 +454,27 @@ struct User <: AbstractParameterNode
     return_factor::Vector{Float64}
     min_level::Vector{Float64}
     priorities::Vector{Int}
+    allocation_optimized::BitVector
+    record::NamedTuple{
+        (
+            :time,
+            :allocation_network_id,
+            :user_node_id,
+            :priority,
+            :demand,
+            :allocated,
+            :abstracted,
+        ),
+        Tuple{
+            Vector{Float64},
+            Vector{Int},
+            Vector{Int},
+            Vector{Int},
+            Vector{Float64},
+            Vector{Float64},
+            Vector{Float64},
+        },
+    }
```

Comment:
I think here the NamedTuple macro may lead to more intuitive syntax: https://docs.julialang.org/en/v1/base/base/#Base.@NamedTuple